src/Cooked/MockChain/BlockChain.hs

```diff
@@ -351,19 +349,19 @@ awaitEnclosingSlot = awaitSlot <=< getEnclosingSlot
 
 -- | Wait a given number of ms from the lower bound of the current slot and
 -- returns the current slot after waiting.
-awaitDurationFromLowerBound :: (MonadBlockChainWithoutValidation m) => Integer -> m Ledger.Slot
-awaitDurationFromLowerBound duration = currentTime >>= awaitEnclosingSlot . (+ fromIntegral duration) . fst
+waitNMSFromSlotLowerBound :: (MonadBlockChainWithoutValidation m, Integral i) => i -> m Ledger.Slot
```

Comment:
IMO `waitMSDurationSlot...`  might seem more self explanatory when used out of context. When I first saw this function, I thought "What is NMS?`.